src/custom hooks/useToggleVideo.js

```diff
@@ -0,0 +1,44 @@
+import { successToast } from "../components/toasts";
+import { usePlaylists } from "../pages/Playlists/playlists-context";
+import {
+  ADD_VIDEO_TO_PLAYLIST,
+  REMOVE_VIDEO_FROM_PLAYLIST,
+} from "../pages/Playlists/playlists-reducer";
+import { API_PLAYLISTS } from "../urls";
+import { useAxios } from "./useAxios";
+
+export function useToggleVideo({ playlistId, video }) {
+  const { playlistsDispatch } = usePlaylists();
+  const {
+    updateData: addVideoToPlaylist,
+    deleteData: removeVideoFromPlaylist,
+    isLoading,
+  } = useAxios(`${API_PLAYLISTS}/${playlistId}`);
+  const toggleVideoInPlaylist = async (isPresent) => {
+    if (!isPresent) {
+      const success = await addVideoToPlaylist(video._id, {});
+      if (success) {
+        successToast(`Added video to playlist`);
+        playlistsDispatch({
+          type: ADD_VIDEO_TO_PLAYLIST,
+          video,
+          playlistId,
+        });
+      }
+    } else {
+      const success = await removeVideoFromPlaylist(video._id);
+      if (success) {
```

Comment:
You can extract these also to a separate function. It would be easier to understand.

Reply:
Again a great suggestion. Updated this